File: Part2/Code/depression.py
#https://pdas.samhsa.gov/saes/substate#

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import glob
from sklearn.feature_extraction.text import CountVectorizer
#cd HW-02

#load data from csv file
all_data_frames=glob.glob("./Part2/Data/*.csv")+glob.glob("./Part2/Data/*.xlsx")
for fname_index, fname in enumerate(all_data_frames):
    print(fname_index, fname)

#A.clean the data named mental_health
mental_health = all_data_frames[9]
mental_health_df = pd.read_excel(mental_health)  # Use the 'mental_health' variable
# show all the columns of the dataframe
# mental_health_df.columns
print(mental_health_df.head())
# # drop the columns of the dataframes that do not belong to mental health data
mental_health_df = mental_health_df.drop(columns=['Drug use disorders (%)', 'Alcohol use disorders (%)'])
print(mental_health_df.head())
#save the dataframe to csv file
mental_health_df.to_csv("./Part2/Code/mental_health.csv", index=False)



# #B.clean the data named eating_disorders
# eating_disorder = all_data_frames[3]
# print(eating_disorder)
# eating_disorder_df = pd.read_csv(eating_disorder) # Use the 'eating_disorder' variablev
# # show all the columns of the dataframe
# print(eating_disorder_df.columns)
# print(eating_disorder_df.head())
# # drop the columns Continent
# eating_disorder_df = eating_disorder_df.drop(columns=['Continent','Population (historical estimates)'])
# print(eating_disorder_df.shape)
# # only keep the rows of the dataframe when the column 'Year' is between 1990-2021
# # eating_disorder_df = eating_disorder_df.loc[eating_disorder_df['Year'] == range(1990,2022)]
# eating_disorder_df = eating_disorder_df[eating_disorder_df['Year'].isin(range(1990,2022))]
# print(eating_disorder_df.shape)
# rename_map={
#     'Entity':'Country',
#     'Code':'Country Code',
#     'Year':'Year',
#     'Eating disorders (share of population) - Sex: Male - Age: Age-standardized': 'Male',
#     'Eating disorders (share of population) - Sex: Female - Age: Age-standardized':'Female',
# }
# eating_disorder_df.rename(columns=rename_map, inplace=True)
# print(eating_disorder_df.head())
# #caculate the mean of the column Male and Female of the dataframe and save it to a new column named All_gender
# eating_disorder_df['All_gender'] = eating_disorder_df[['Male', 'Female']].mean(axis=1)
# print(eating_disorder_df.head())
# #save the dataframe to csv file
# eating_disorder_df.to_csv("./Part2/Code/eating_disorder.csv", index=False)

# #C.clean the data named discomfort_speaking_anxiety_depression_2020 and Contries_Continents
# Contries_Continents=all_data_frames[0]
# Contries_Continents_df = pd.read_csv(Contries_Continents) 
# print(Contries_Continents_df.head())
# print(Contries_Continents_df.shape)
# discomfort_speaking_anxiety_depression_2020=all_data_frames[1]
# discomfort_speaking_anxiety_depression_2020_df = pd.read_csv(discomfort_speaking_anxiety_depression_2020)
# print(discomfort_speaking_anxiety_depression_2020_df.head())
# print(discomfort_speaking_anxiety_depression_2020_df.columns)
# print(discomfort_speaking_anxiety_depression_2020_df.shape)
# #rename the column 'Entity' to 'Country' and 'Share - Question: mh5 - Someone local comfortable speaking about anxiety/depression with someone they know - Answer: Not at all comfortable - Gender: all - Age_group: all' to 'Not_Comfortable'
# discomfort_speaking_anxiety_depression_2020_df.rename(columns={'Entity':'Country','Share - Question: mh5 - Someone local comfortable speaking about anxiety/depression with someone they know - Answer: Not at all comfortable - Gender: all - Age_group: all':'Not_Comfortable'}, inplace=True)
# # merge the two dataframes
# discomfort_in_continent= discomfort_speaking_anxiety_depression_2020_df.merge(Contries_Continents_df, how='left', left_on='Country', right_on='Country')
# print(discomfort_in_continent.head())
# ##save the dataframe to csv file
# discomfort_in_continent.to_csv("./Part2/Code/discomfort_in_continent.csv", index=False)

File: Part2/Code/depression.r
# Load the required libraries
library(readxl)
library(dplyr)

# List all data files
data_files <- list.files(path = "./Part2/Data", full.names = TRUE)

#A. Load mental_health data
mental_health_file <- data_files[10] # Assuming mental health data is at index 10
mental_health_df <- read_excel(mental_health_file)
head(mental_health_df)

# Drop columns not related to mental health
mental_health_df <- mental_health_df %>%
  select(-c(`Drug use disorders (%)`, `Alcohol use disorders (%)`))
head(mental_health_df)

# Save the cleaned data to CSV
write.csv(mental_health_df, file = "./Part2/Code/mental_health_r.csv", row.names = FALSE)


# # B.Load eating_disorders data
# eating_disorder_file <- data_files[4] # Assuming eating disorders data is at index 4
# eating_disorder_df <- read.csv(eating_disorder_file)
# head(eating_disorder_df)

# # Drop columns not needed
# eating_disorder_df <- eating_disorder_df %>%
#   select(-c(Continent, `Population (historical estimates)`))
# head(eating_disorder_df)

# # Filter data for years between 1990 and 2021
# eating_disorder_df <- eating_disorder_df %>%
#   filter(Year >= 1990 & Year <= 2021)
# head(eating_disorder_df)

# # Rename columns
# eating_disorder_df <- eating_disorder_df %>%
#   rename(Country = Entity, `Country Code` = Code, Year, Male, Female)
# head(eating_disorder_df)

# # Calculate the mean of Male and Female columns
# eating_disorder_df <- eating_disorder_df %>%
#   mutate(All_gender = rowMeans(.[c("Male", "Female")], na.rm = TRUE))
# head(eating_disorder_df)

# # Save the cleaned data to CSV
# write.csv(eating_disorder_df, file = "./Part2/Code/eating_disorder_r.csv", row.names = FALSE)



#C. Load Countries_Continents data
# countries_continents_file <- data_files[1] # Assuming Countries_Continents data is at index 1
# countries_continents_df <- read.csv(countries_continents_file)
# head(countries_continents_df)

# # Load discomfort_speaking_anxiety_depression_2020 data
# discomfort_data_file <- data_files[2] # Assuming discomfort_speaking_anxiety_depression_2020 data is at index 2
# discomfort_data_df <- read.csv(discomfort_data_file)
# head(discomfort_data_df)

# # Rename columns
# discomfort_data_df <- discomfort_data_df %>%
#   rename(Country, Not_Comfortable = `Share - Question: mh5 - Someone local comfortable speaking about anxiety/depression with someone they know - Answer: Not at all comfortable - Gender: all - Age_group: all`)
# head(discomfort_data_df)

# # Merge the two dataframes
# discomfort_in_continent <- merge(discomfort_data_df, countries_continents_df, by = "Country", all.x = TRUE)
# head(discomfort_in_continent)

# # Save the cleaned data to CSV
# write.csv(discomfort_in_continent, file = "./Part2/Code/discomfort_in_continent_r.csv", row.names = FALSE)
